openshift-201/pipelines.md

```diff
@@ -0,0 +1,232 @@
+# Introduction to OpenShift Pipelines
+
+## Objectives:
+
+After completing this section, you should have an understanding of OpenShift Pipelines, know how to create and start a pipeline, and how to migrate from Jenkins
+
+## Prerequisites
+This lab is based on the `pipeline-templates` which has it's own set of [prerequisites](https://github.com/bcgov/pipeline-templates/tree/main/tekton#prerequisites) and [installation instructions](https://github.com/bcgov/pipeline-templates/tree/main/tekton#installation).
+
+Please follow those instructions before continuing on.
+
+**NOTE:** You must be a member of the `bcgov` GitHub org to obtain a valid token from SonarCloud.  If you haven't already please create a token in [SonarCloud](https://sonarcloud.io/) and make sure you've set the token in the `secrets.ini` file during the installation instructions above.
+
+## Tasks
+`Tasks` are the building blocks of a pipeline and consists of sequentially executed steps. It is essentially a function of inputs and outputs. `Tasks` are reusable and can be used in multiple Pipelines.
+
+`Steps` are a series of commands that are sequentially executed by the task and achieve a specific goal, such as building an image. Every task runs as a pod, and each step runs as a container within that pod. Because steps run within the same pod, they can access the same volumes for caching files, config maps, and secrets.
+
+
+The following example shows the [`t-generate-id`](https://github.com/bcgov/pipeline-templates/blob/main/tekton/base/tasks/generate-id.yaml) task.
```

Comment:
Should the task name match the screenshot?

```suggestion
The following example shows the [`generate-id`](https://github.com/bcgov/pipeline-templates/blob/main/tekton/base/tasks/generate-id.yaml) task.
```

Reply:
Fixed the name in the image since when the script runs in the pipeline-template install it prefixes the name with `t-` and thus wanted this to match with what they will see in the cluster